static/src/js/reducers/ui.js

```diff
@@ -195,26 +200,74 @@ const uiReducer = (state = initialState, action) => {
       }
     }
     case EXPORT_STARTED: {
-      return {
-        ...state,
-        export: {
-          isExportRunning: {
-            ...state.export.isExportRunning,
-            [action.payload]: true
+      const { type, format } = action.payload
+
+      if (type === 'collection') {
+        return {
+          ...state,
+          export: {
+            isExportRunning: {
+              ...state.export.isExportRunning,
+              collection: {
+                ...state.export.isExportRunning.collection,
+                [format]: true
+              }
+            }
+          }
+        }
+      }
+
+      if (type === 'granule') {
+        return {
+          ...state,
+          export: {
+            isExportRunning: {
+              ...state.export.isExportRunning,
+              granule: {
+                ...state.export.isExportRunning.granule,
+                [format]: true
+              }
+            }
           }
         }
       }
+
+      return state
     }
     case EXPORT_FINISHED: {
-      return {
-        ...state,
-        export: {
-          isExportRunning: {
-            ...state.export.isExportRunning,
-            [action.payload]: false
+      const { type, format } = action.payload
+
+      if (type === 'collection') {
```

Comment:
Can we use the type as a variable on lines 246 and 261 to avoid the duplicate code?